### PR TITLE
Completes ieee1788-constructors.jl

### PR DESCRIPTION
**Pull request by krish8484**

### CODE REVIEW

**Comment by krish8484** [comment]
ITF1788 Tests
Master
```
julia> DecoratedInterval(2,1) == nai()
```
crashes 

This PR
```
julia> DecoratedInterval(2,1) == nai()
true
```

**Comment by krish8484** [comment]
Master
```
julia> decoration(@decorated("3.56?1_def"))
com::DECORATION = 4
```
This PR
```
julia> decoration(@decorated("3.56?1_def"))
def::DECORATION = 2
```

File: src/decorations/functions.jl [hunk 1]
```diff
@@ -34,6 +43,11 @@ end
 in(x::T, a::DecoratedInterval) where T<:Real = in(x, interval_part(a))
 
 
+function ==(x::DecoratedInterval, y::DecoratedInterval)
+    isnai(x) && isnai(y) && return true
```
Comment (dpsanders) [inline src/decorations/functions.jl:47]: Should two `nai` intervals be equal? Is this required by the standard?  E.g. `NaN == NaN`  is false.

File: src/decorations/functions.jl [hunk 2]
```diff
@@ -34,6 +43,11 @@ end
...
+    return (==(interval_part(x), interval_part(y)))
```
Comment (dpsanders) [inline src/decorations/functions.jl:48]: Now would be a good time to replace `interval_part` by `interval` everywhere.

Comment (dpsanders) [reply src/decorations/functions.jl:48]: We should probably deprecate `interval_part`. (Use `@deprecate`)

File: test/ITF1788_tests/ITF1788_tests.jl [hunk 3]
```diff
@@ -1,4 +1,4 @@
-
+include("ieee1788-constructors.jl")
```
Comment (dpsanders) [inline test/ITF1788_tests/ITF1788_tests.jl:1]: Is this a new file that you wrote or part of the ITF1788 suite?

File: test/ITF1788_tests/ieee1788-constructors.jl [hunk 4]
```diff
@@ -0,0 +1,115 @@
+#=
+ 
+ Test Cases for interval constructors from IEEE Std 1788-2015
```
Comment (dpsanders) [inline test/ITF1788_tests/ieee1788-constructors.jl:3]: Ah, sorry, I see that this answers the previous question!

**Comment by dpsanders** [comment]
LGTM, thanks a lot! This stuff is fiddly and annoying -- great job!

File: src/decorations/functions.jl (hunk 1 above)
Comment (krish8484) [reply src/decorations/functions.jl:47]: `DecoratedInterval(2, 1) == nai()`  is true according to the standard, so `NaN == NaN` must also be true.

Comment (lbenet) [reply src/decorations/functions.jl:47]: Ups, my mistake, I didn't see the next line. Sorry, you are right.

**Comment by krish8484** [comment]
https://github.com/JuliaIntervals/IntervalArithmetic.jl/projects/2#column-9620096

**Comment by dpsanders** [comment]
I'm getting
```
julia> x = @decorated("3.56?1_def")
[3.54999, 3.57001]

julia> decoration(x)
com::DECORATION = 4
```

**Comment by dpsanders** [comment]
LGTM!

**Pull request merged**